src/libcollections/binary_heap.rs

```diff
@@ -967,6 +974,13 @@ pub struct Iter<'a, T: 'a> {
     iter: slice::Iter<'a, T>,
 }
 
+#[stable(feature = "collection_debug", since = "1.15.0")]
+impl<'a, T: 'a> fmt::Debug for Iter<'a, T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad("BinaryHeap::Iter { .. }")
```

Comment:
What's the plan here about showing the elements? It needs specialization to add the case for `T: fmt::Debug` later, and I don't think we want the complexity of having both impls (without debug bound and with) for each struct.